### ISSUE TITLE
Joint constraints still produce angles in the constrained DOFs

### ISSUE THREAD

**Issue by zicodasilva** [alknemeyer/physical_education#9]
So I've noticed that the `robot.plot(...)` produces position states with some non-zero values for the constrained axes. I wouldn't expect it to be perfect and possibly `+-0.1` rad would have been ok. But the image below, for a revolute joint (which allows movement in the pitch angle), we are getting values close to `0.5` rad for the constrained DOFs.
<img width="425" alt="image" src="https://user-images.githubusercontent.com/8395831/174871376-20601ac9-4235-43c6-9dd3-6a82fc3db811.png">
Funny enough, I don't see this "violation" visually, even though it is like 30 degrees. Not sure what to make of this, but thought to just put this here, to start a discussion.

**Comment by zicodasilva** [comment]
The first 5 finite elements, is possibly so quick that I don't really notice it.

**Comment by alknemeyer** [comment]
As far as I remember, the constraints are for the angles between links. If so, a non-zero angle in inertial coordinates is possible and expected. Could you check the relative angles?

Otherwise, could you send a video or describe the movement in more detail?

**Comment by zicodasilva** [comment]
Yes, they are angles between links. That is correct, I should be providing the relative angles here. Probably a more useful property of the plotting function anyway. I'm a bit concerned about the change in yaw and roll angles in the above plot, but that could happen if the body is rolling and changing direction (something for me to investigate).

I will repurpose this issue to implement relative angle plots as an option when plotting states of the robot.
